code/datums/elements/supermatter_crystal.dm

```diff
@@ -0,0 +1,343 @@
+				span_danger("Oops! The [item] flashes out of existence on contact with \the [atom_source], taking your arm with it! That was clumsy of you!"))
+			playsound(atom_source, 'sound/effects/supermatter.ogg', 150, TRUE)
+			consume(atom_source, dust_arm)
+			qdel(item)
+			return
+		if(cig.lit || user.combat_mode)
+			user.visible_message(span_danger("A hideous sound echoes as [item] is ashed out on contact with \the [atom_source]. That didn't seem like a good idea..."))
+			playsound(atom_source, 'sound/effects/supermatter.ogg', 150, TRUE)
+			consume(atom_source, item)
+			radiation_pulse(atom_source, max_range = 3, threshold = 0.1, chance = 50)
+			return
+		else
+			cig.light()
+			user.visible_message(span_danger("As [user] lights \their [item] on \the [atom_source], silence fills the room..."),\
+				span_danger("Time seems to slow to a crawl as you touch \the [atom_source] with \the [item].</span>\n<span class='notice'>\The [item] flashes alight with an eerie energy as you nonchalantly lift your hand away from \the [atom_source]. Damn."))
+			playsound(atom_source, 'sound/effects/supermatter.ogg', 50, TRUE)
+			radiation_pulse(atom_source, max_range = 1, threshold = 0, chance = 100)
+			return
+	if(istype(item, /obj/item/scalpel/supermatter))
+		var/obj/item/scalpel/supermatter/scalpel = item
+		INVOKE_ASYNC(src, .proc/use_scalpel, scalpel, user, atom_source)
+
+	else if(user.dropItemToGround(item))
+		user.visible_message(span_danger("As [user] touches \the [atom_source] with \a [item], silence fills the room..."),\
+			span_userdanger("You touch \the [atom_source] with \the [item], and everything suddenly goes silent.</span>\n<span class='notice'>\The [item] flashes into dust as you flinch away from \the [atom_source]."),\
+			span_hear("Everything suddenly goes silent."))
+		user.investigate_log("has been attacked ([item]) by [key_name(user)]", INVESTIGATE_ENGINE)
+		consume(atom_source, item)
+		playsound(get_turf(atom_source), 'sound/effects/supermatter.ogg', 50, TRUE)
+
+		radiation_pulse(atom_source, max_range = 3, threshold = 0.1, chance = 50)
+
+	else if(atom_source.Adjacent(user)) //if the item is stuck to the person, kill the person too instead of eating just the item.
+		var/vis_msg = span_danger("[user] reaches out and touches [atom_source] with [item], inducing a resonance... [item] starts to glow briefly before the light continues up to [user]'s body. [user.p_they(TRUE)] bursts into flames before flashing into dust!")
+		var/mob_msg = span_userdanger("You reach out and touch [atom_source] with [item]. Everything starts burning and all you can hear is ringing. Your last thought is \"That was not a wise decision.\"")
+		dust_mob(source, user, vis_msg, mob_msg)
+
+/datum/element/supermatter_crystal/proc/tool_hit(datum/source, mob/user, obj/item/tool)
+	SIGNAL_HANDLER
+	if(tool.tool_behaviour == TOOL_WRENCH && tool_act_callback)
```

Comment:
you don't need to check for wrench tool behavior.
this signal is only registered on `COMSIG_ATOM_TOOL_ACT(TOOL_WRENCH)`, so only items with the `TOOL_WRENCH` behavior will reach here with a signal